via/static/scripts/video_player/components/test/ToastMessages-test.js

```diff
@@ -0,0 +1,109 @@
+import { mount } from 'enzyme';
+
+import { delay } from '../../../test-util/wait';
+import ToastMessages from '../ToastMessages';
+
+describe('ToastMessages', () => {
+  const toastMessages = [
+    {
+      id: 'abc',
+      type: 'success',
+      message: 'Hello world',
+    },
+    {
+      id: 'def',
+      type: 'success',
+      message: 'Foobar',
+    },
+    {
+      id: 'ghi',
+      type: 'error',
+      message: 'Something failed',
+    },
+  ];
+  let fakeOnMessageDismiss;
+
+  beforeEach(() => {
+    fakeOnMessageDismiss = sinon.stub();
+  });
+
+  function createToastMessages(toastMessages, setTimeout) {
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+
+    return mount(
+      <ToastMessages
+        messages={toastMessages}
+        onMessageDismiss={fakeOnMessageDismiss}
+        _setTimeout={setTimeout}
+      />,
+      { attachTo: container }
+    );
+  }
+
+  it('renders a list of toast messages', () => {
+    const wrapper = createToastMessages(toastMessages);
+    assert.equal(wrapper.find('ToastMessageItem').length, toastMessages.length);
+  });
+
+  toastMessages.forEach((message, index) => {
+    it('dismisses messages when clicked', async () => {
+      const wrapper = createToastMessages(toastMessages);
+
+      wrapper.find('Callout').at(index).props().onClick();
+      // onMessageDismiss is not immediately called. Transition has to finish
+      assert.notCalled(fakeOnMessageDismiss);
+
+      // Once dismiss transition has finished, onMessageDismiss is called
+      await delay(350); // "out" transition lasts 300ms
```

Comment:
We definitely need to find a way to make this run faster in the tests. Also this could cause test flakes if the animation runs more slowly than expected for any reason.

You mentioned using a custom Tailwind config as one option. That would be high fidelity, but could turn out to be brittle. Some other ideas:

- Dispatch fake `animationend` events at the DOM elements
- Add a test seam to allow animation ends to be triggered sooner another way

Reply:
Thanks for the suggestions. That didn't occur to me, and I think it should be possible to get something workable with that.